### PR TITLE
Add  beforeUpdateQuantity to BeforeLineItemQuantityChangedEvent to allow plugins to understand if quantity increases or decreases

### PR DESCRIPTION
**Pull request by luminalpark**
### 1. Why is this change necessary?
`BeforeLineItemQuantityChangedEvent `should expose a `beforeUpdateQuantity` property , to allow plugins to understand if quantity increases or decreases. 

### 2. What does this change do, exactly?
Adds a new `beforeUpdateQuantity` property to `BeforeLineItemQuantityChangedEvent` to allow plugins to understand if quantity is being increased or decreased

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by aragon999** [comment]
While I understand the purpose of this change, this will break all plugins which rely on the fact, that `$lineItem->getQuantity()` will yield the changed quantity.

**Comment by luminalpark** [comment]
> While I understand the purpose of this change, this will break all plugins which rely on the fact, that `$lineItem->getQuantity()` will yield the changed quantity.

Thanks a lot for sharing your view on my first PR! You're right.
The event dispatch could remain unmodified, I could add a `beforeUpdateQuantity` to the event to keep the desired functionality. What do you think?

**Comment by luminalpark** [comment]
I reverted part of the code, and added a beforeUpdateQuantity property to let plugins handle the delta: `$event->getBeforeUpdateQuantity() - $event->getLineItem()->getQuantity()`

File: src/Core/Checkout/Cart/Event/BeforeLineItemQuantityChangedEvent.php [hunk 1]
```diff
@@ -28,10 +33,12 @@ class BeforeLineItemQuantityChangedEvent implements ShopwareSalesChannelEvent, C
     protected $salesChannelContext;
 
     public function __construct(
+        int $beforeUpdateQuantity,
```
Comment (keulinho) [inline src/Core/Checkout/Cart/Event/BeforeLineItemQuantityChangedEvent.php]: this is a breaking change, maybe add a separate setter for that or directly use a public property?

File: src/Core/Checkout/Cart/Event/BeforeLineItemQuantityChangedEvent.php [hunk 2]
```diff
@@ -12,6 +12,11 @@
 #[Package('checkout')]
 class BeforeLineItemQuantityChangedEvent implements ShopwareSalesChannelEvent, CartEvent
 {
+    /**
+     * @var int
+     */
+    protected $beforeUpdateQuantity;
```
Comment (keulinho) [inline src/Core/Checkout/Cart/Event/BeforeLineItemQuantityChangedEvent.php]: Please use typed properties for all new code

**Comment by Bird87ZA** [comment]
@luminalpark Thanks for opening the PR.

Could you also write tests for your changes?

**Pull request closed without merging**